src/LCRmeter-Metrohm_AutolabPGSTAT/main.py

```diff
@@ -0,0 +1,339 @@
+        self.dc_bias = float(parameter["ValueBias"])
+        self.frequency = float(parameter["Frequency"])
+        self.speed = self.speeds[parameter["Integration"]]
+        self.measure_range = self.measurement_ranges[parameter["Range"]]
+        self.number_of_cycles = int(parameter["Average"])
+
+    def connect(self) -> None:
+        """Connect to the Metrohm Autolab LCRmeter."""
+        if import_failed:
+            msg = "Could not import required libraries."
+            raise ImportError(msg)
+
+        self.autolab = Instrument()
+        self.autolab.AutolabConnection.EmbeddedExeFileToStart = self.adx_path
+        self.autolab.set_HardwareSetupFile(self.hardware_setup_file)
+
+        print("Connecting to Autolab...")
+        # TODO: handle error if Nova is still running or not connected to the device
+        self.autolab.Connect()
+        print("Connected to Autolab.")
+
+        self.Fra = self.autolab.Fra
+        self.Ei = self.autolab.Ei
+
+    def disconnect(self) -> None:
+        """Disconnect from the Metrohm Autolab LCRmeter."""
+        self.autolab.Disconnect()
+
+    def initialize(self) -> None:
+        """Initialize the Metrohm Autolab LCRmeter."""
+
+    def deinitialize(self) -> None:
+        """Reset device and close connection."""
+
+    def configure(self) -> None:
+        """Set bias and measurement parameters with start values from GUI."""
+        # Initial Measurement Parameters
+        self.set_frequency(self.frequency)
+        self.set_dc_bias(self.dc_bias)
+        self.set_ac_voltage(self.ac_amplitude)
+
+        # General Measurement Parameters
+        self.Fra.WaveType = self.Fra.WaveType.Sine
+
+        # If needed, this parameter could be added to the GUI
+        self.Fra.MinimumIntegrationTime = 0.125
+
+        self.Fra.MinimumIntegrationCycles = self.number_of_cycles
+        self.set_measure_range(self.measure_range)
+        self.set_speed(self.speed)
+
+        # Prepare device
+        self.Ei.EnableDsgInput = True
+        self.Ei.CellOnOff = self.Ei.EICellOnOff.On
+
+        InstrumentExtensions.SwitchFraOn(self.autolab)
+        time.sleep(1)  # wait to stabilize
+
+    def unconfigure(self) -> None:
+        """Reset device."""
+        self.Ei.CellOnOff = self.Ei.EICellOnOff.Off
+        self.Ei.EnableDsgInput = False
+
+    def apply(self) -> None:
+        """Apply settings."""
+        if self.sweepmode != "None":
+            sweep_value = float(self.value)
+            self.handle_set_value(self.sweepmode, sweep_value)
+
+        if self.stepmode != "None":
+            step_value = float(self.stepvalue)
+            self.handle_set_value(self.stepmode, step_value)
+
+    def measure(self) -> None:
+        """Retrieve Impedance results from device."""
+        # Run the measurement
+        self.Fra.Start()
+
+        # Retrieve measured values
+        self.resistance, self.reactance = self.measure_resistance_reactance()
```

Comment:
would it make sense to move this to request_result() or read_result() semantic functions?
Probably only makes a difference if self.Fra.Start() is non-blocking